spec/formatting.md

```diff
@@ -230,65 +231,77 @@ the following steps are taken:
 - A _function_ _annotation_ fails to resolve.
 - A _private-use_ _annotation_ is unsupported by the implementation or if
   a _private-use_ _annotation_ fails to resolve.
 - The _expression_ has a _reserved_ _annotation_.
 
 The _fallback value_ depends on the contents of the _expression_:
 
-- _expression_ with _literal_ _operand_: U+007C VERTICAL LINE `|`
-  followed by the value of the Literal,
-  and then by U+007C VERTICAL LINE `|`.
-  The same representation is used for both _quoted_ and _unquoted_ values.
+- _expression_ with _function_ _annotation_ and no _operand_:
+  the _annotation_ starting sigil followed by its _identifier_
 
-  > Examples: `|your horse|`, `|42|`
+  > Examples:
+  > In a context where `:func` fails to resolve, `{:func}` resolves to the _fallback value_ `:func`.
+  > In a context where `:ns:func` fails to resolve, `{:ns:func}` resolves to the _fallback value_ `:ns:func`.
 
-- _expression_ with _variable_ _operand_: U+0024 DOLLAR SIGN `$`
-  followed by the _variable_ _name_ of the _operand_
+- _expression_ with non-_function_ _annotation_ and no _operand_:
+  - If the _annotation_ consists of a _sigil_-prefixed _identifier_
+    optionally followed by a whitespace-prefixed tail:
+    the _annotation_ starting sigil followed by that _identifier_
 
-  > Example: `$user`
+    > Examples:
+    > In any context, `{@reserved}` and `{@reserved |...|}` both resolve to the _fallback value_ `@reserved`.
 
-- _expression_ with no _operand_:
-  the _function_ starting sigil followed by its _name_
+  - Otherwise: the U+FFFD REPLACEMENT CHARACTER `�`
 
-  > Examples: `:platform`, `+tag`, `-tag`
+    > Examples:
+    > In any context, `{@reserved\\}` and `{? foo ?}` both resolve to the _fallback value_ `�`.
 
-- Otherwise: The U+FFFD REPLACEMENT CHARACTER `�` character
+- _expression_ with _literal_ _operand_:
+  U+007C VERTICAL LINE `|`
+  followed by the value of the _literal_
+  with escaping applied to U+005C REVERSE SOLIDUS `\` and U+007C VERTICAL LINE `|`,
+  and then by U+007C VERTICAL LINE `|`.
+  The same representation is used for both _quoted_ and _unquoted_ values.
 
-_Option_ names and values are not included in the _fallback value_.
+  > Examples:
+  > In a context where `:func` fails to resolve,
+  > `{42 :func}` resolves to the _fallback value_ `|42|` and
+  > `{|C:\\| :func}` resolves to the _fallback value_ `|C:\\|`.
+  > In any context, `{|| @reserved}` resolves to the _fallback value_ `||`.
 
-When an error occurs in an _expression_ with a _variable_ _operand_
-and the _variable_ refers to a local _declaration_,
-the _fallback value_ is formatted based on the _expression_
-on the right-hand side of the _declaration_,
-rather than the _expression_ in the _selector_ or _pattern_.
+- _expression_ with _variable_ _operand_ (with or without an _annotation_):
+  - If the _variable_ fails to resolve, or resolves to a value that is not a _fallback value_:
```

Comment:
I found this confusing. You have to read everything here, including the next case, to understand how a fallback value could get into the variable. Unless you know about the transitive case, this appears to say "all variables". I would suggest putting the variable fails to resolve case first, then the transitive case (starting line 291) and end with an "otherwise" case for variables that resolve correctly but whose expressions fail.

Yes, that will be slightly repetitive in the last case.